docs/decisions/0002-lerna-monorepo.rst

```diff
@@ -0,0 +1,108 @@
+
+To get around this issue of common/shared dependencies, we can rely on how NPM finds installed packages. If a package does not exist in ``node_modules`` for an individual package, NPM will look in ``node_modules`` further up the directory tree until it finds the package, or gets to the root of the repository. 
+
+By installing these common dependencies at the root package.json file, they will be accessible to any package in the monorepo to ensure there is only one copy of them used throughout. These dependencies are still noted in each individual package.json file as a peer dependency but not as a dev dependency since they are already installed in ``node_modules`` at the root of the repository.
+
+As such, we should pay extra attention to managing dependencies in each packages, making informed decisions about whether a dependency should be included in an individual package's package.json file or the package.json file at the root of the repository.
+
+================================
+Running CI for branches and pull requests
+================================
+
+When a branch is pushed or a pull request is created on Github, CI will continue to run tests, linting, and other necessary build steps to ensure changes did not cause regressions. The difference now is that we rely on Lerna to run commands across all packages that have changed since the last release. This will ensure when a change is made to a package, the CI will be sure to run any associated tests for that specific package, including any of its dependencies and dependents across the monorepo. Output from each command Lerna runs concurrently across all packages are streamed in each Github Action step.
+
+================================
+NPM releases with Lerna
+================================
+
+Publishing packages to NPM remains a job of Github Actions, however, releases will no longer happen using ``semantic-release`` directly. Instead, we will rely on the ``lerna publish`` command to determine which packages have changed since the last merged commit and release, and only publish those. 
+
+Lerna is configured to use independent versioning with conventional commits, as opposed to keeping all package versions in sync. This will allow us to publish packages independently of each other, rather than in lock-step like some JS monorepos do (e.g., Babel, React). The benefit of independent versioning is that consuming applications won't get erroneous upgrades when code unrelated to their use case is updated in the monorepo. For example, say the ``edx/frontend-app-learning`` micro-frontend consumes ``@edx/frontend-enterprise-utils``. Without independent versioning, if we released a new version of ``@edx/frontend-enterprise-catalog-search``, then ``edx/frontend-app-learning`` would have an upgrade available when nothing about the package changed.
+
```

Comment:
If a change to utils module is made and it only breaks logistration but not catalog-search, for example, will the deployment of utils module also be gated, or would it deploy regardless?

Reply:
The Github Actions CI runs tests for each package that was changed since the last release and any of its dependencies and dependents. So, if we change `frontend-enterprise-utils` and break something, CI _should_ fail if it causes `frontend-enterprise-catalog-search` and/or `frontend-enterprise-logistration` tests to fail since both of those packages are dependents on `frontend-enterprise-utils`.

But yeah, this is actually something I still need to explicitly test to see it holds up to what should conceptually happen. I will try to replicate this situation locally and report back in the morning!

Reply:
Confirmed. I caused a bug in `@edx/frontend-enterprise-utils` in a function that's imported by `@edx/frontend-enterprise-catalog-search`, and the `npm run ci:test` ended up running both `@edx/frontend-enterprise-catalog-search` and `@edx/frontend-enterprise-logistration` (since both depend on `@edx/frontend-enterprise-utils`), and the test suite for `@edx/frontend-enterprise-catalog-search` failed due to the regression made, which would fail CI and prevent publishing of the new version of `@edx/frontend-enterprise-utils`.